mmdet/core/anchor/anchor_generator.py

```diff
@@ -245,22 +251,23 @@ def single_level_grid_priors(self, featmap_size, level_idx, device='cuda'):
         Args:
             featmap_size (tuple[int]): Size of the feature maps.
             level_idx (int): The index of corresponding feature map level.
+            dtype (obj:`torch.dtype`): Date type of points.Defaults to
+                ``torch.float32``.
             device (str, optional): The device the tensor will be put on.
                 Defaults to 'cuda'.
 
         Returns:
             torch.Tensor: Anchors in the overall feature maps.
         """
 
-        base_anchors = self.base_anchors[level_idx].to(device)
+        base_anchors = self.base_anchors[level_idx].to(device).to(dtype)
         feat_h, feat_w = featmap_size
         stride_w, stride_h = self.strides[level_idx]
-        shift_x = torch.arange(0, feat_w, device=device) * stride_w
-        shift_y = torch.arange(0, feat_h, device=device) * stride_h
+        shift_x = torch.arange(0, feat_w, device=device).to(dtype) * stride_w
```

Comment:
can torch.arange set dtype？

Reply:
I refer to 
https://github.com/open-mmlab/mmdetection/blob/003f520cad88be52bd5fd44cdcab2d7ec45dd551/mmdet/models/dense_heads/anchor_free_head.py#L297
Please @RunningLeon take a look
I will add a similar comment here.

Reply:
better add a comment for explaination.